r/src/filesystem.cpp

```diff
@@ -287,6 +287,9 @@ std::shared_ptr<fs::S3FileSystem> fs___S3FileSystem__create(
     std::string session_name = "", std::string external_id = "", int load_frequency = 900,
     std::string region = "", std::string endpoint_override = "", std::string scheme = "",
     bool background_writes = true) {
+  // We need to ensure that S3 is initialized before we start messing with the
+  // options
+  StopIfNotOk(fs::EnsureS3Initialized());
```

Comment:
We might not need this to be _this_ high — I believe the segfault was `s3_opts = fs::S3Options::Defaults();` below, but it shouldn't hurt to have it up here